src/implementation.md

```diff
@@ -74,6 +74,29 @@ We will use [binaryen](https://github.com/WebAssembly/binaryen) to run WebAssemb
 
 - [cpp-libp2p](https://github.com/libp2p/cpp-libp2p) will be used as libp2p implementation. We will extend these libraries to support building with the Conan package manager. Also we will add support of building with Emscripten and websocket clients.
 
+## Light Client Component Interactions
+
+The client will perform light-weight interactions with components (e.g., GRANDPA) by way of two groups of messaging:
+- notification messages
+- request/response messages
+
+Notifications make use of the following protocols:
+- /dot/block-announces/1
+- /dot/transactions/1
+- /paritytech/grandpa/1
+
+Block announces are intentionally sent only to light clients, and not to full nodes. Block announces coming from light clients are useless to full nodes, as they can't download the block body (which they need) from that light client.
```

Comment:
I didn't get the last phrase(
Light client shouldn't produce new blocks so it can't make block announces if I understood it right.

Reply:
well, I was scouring the smoldot source for references to difference between full node and light client in the source -- as you've mentioned, that's problematic, because it's not always clear.  But I found this:
https://github.com/paritytech/smoldot/blob/main/bin/light-base/src/sync_service/standalone.rs#L765
But I'll go over the entire, near the line of code, better understand the context, and do a bit of rewrite on the text I have.

Reply:
I cut the second sentence.
I also added two protocols to the "Protocol Support" section
We can discuss, and possibly amend, when I create the proof-read branch (final issue)

Reply:
Oh, see. Maybe it could be better to change the wording to "we are not supposing to implement sending block announces from light client because they are useless to full nodes, as they can't download the block body (which they need) from that light client"

Reply:
good... will add change to my current 'proof-read' branch